packages/agent/src/cbor.ts

```diff
@@ -125,5 +125,9 @@ export function decode<T>(input: ArrayBuffer): T {
     },
   });
 
-  return decoder.decodeFirst(buffer);
+  try {
+    return decoder.decodeFirst(buffer);
+  } catch(e: unknown) {
+    throw new Error(`Failed to decode CBOR: ${e}, input: ${toHex(buffer)}`);
```

Comment:
Out of curiosity, aren't you gonna get something like `Failed to decode CBOR: [object Object]...` - i.e. `e` is always a string?

Reply:
Oh, thanks. I think having the `input: ...` part is definitely an improvement though.
So, what's the simplest way of converting `unknown` to something readable?

Reply:
Not sure what's the best practice or if there is an elegant solution.

In Oisy I got such a parser:

```
export const errorDetailToString = (err: unknown): string | undefined =>
	typeof err === 'string'
		? (err as string)
		: err instanceof Error
			? (err as Error).message
			: 'message' in (err as unknown as { message: string })
				? (err as unknown as { message: string }).message
				: undefined;
```

Another option is maybe to create an error type that extends the error and bubble the new types with the information you need plus the original error. Just thinking out at loud though.

But, again, maybe `e: unknown` is actually always `e: string`, don't know, was just a curiosity question in case not.

Reply:
>But, again, maybe e: unknown is actually always e: string, don't know, was just a curiosity question in case not.

That is definitely not the case, because one of the errors we're expecting to catch here is [this one](https://github.com/dignifiedquire/borc/blob/1bd231963344d8ce2dfb9d9bcf04ea29ff4875de/src/decoder.js#L566). Thanks for pointing that out.

@krpeacock: is there a utility for that in `agent-js` somewhere already? Then I'll adjust it.